internal/profile/files.go

```diff
@@ -0,0 +1,53 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package profile
+
+import (
+	"io/ioutil"
+
+	"github.com/pkg/errors"
+)
+
+// NewConfig is a generic function type to return a new Managed config
+type NewConfig = func(profileName string, profilePath string) (*simpleFile, error)
+
+// simpleFile defines a file that's managed by the profile system
+// and doesn't require any  rendering
+type simpleFile struct {
```

Comment:
if the `simpleFile` is unexported, I assume that all its methods don't have to be exported ?